src/Events/Category_Colors/Migration/Logger.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Handles logging for the category color migration process.
+ * Provides structured logging for errors, warnings, and informational messages.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Events\Category_Colors\Migration
+ */
+
+namespace TEC\Events\Category_Colors\Migration;
+
+/**
+ * Class Logger
+ * Manages logging throughout the migration process.
+ * Supports error, warning, and informational logs to track migration progress.
+ * Logs can be retrieved at any stage to assist with debugging.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Category_Colors\Migration
+ */
+class Logger {
```

Comment:
I'd rather see a logger implemnting PSR.

The loggers are designed as log and forget in order to not put the memory usage in danger. Storing all the log entries of the request could be more problmatic.

That would be a cool class in TCMN which would file the action `tribe_log` which we already use for logging accross our plugins.